src/nvim/terminal.c

```diff
@@ -313,6 +313,10 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
     return;
   }
 
+  if (height == 0 || width == 0) {
+    return;
+  }
```

Comment:
Maybe we should fix this on two fronts : 
  * window.c should not call terminal_resize when it isn't initialized completely yet
  * terminal.c should not let a height or width of 0 pass to vterm_set_size

The second case shouldn't happen, but it's a safeguard for if/when something would change in window.c. (anything with a 400-line function is a problem waiting to happen)

I would still call `win_new_width` from `win_split_ins`, since that does a bunch of other things too. But `win_new_width` could do the check for zero-height

commit added.